lib/gui/lvgl/lvgl.c

```diff
@@ -93,16 +93,20 @@ static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 
 	display_get_capabilities(display_dev, &cap);
 
-	if (cap.x_resolution <= CONFIG_LVGL_HOR_RES_MAX) {
+	if (cap.current_orientation == DISPLAY_ORIENTATION_NORMAL ||
```

Comment:
Drivers are already reporting correct width/height when rotated, so why do this here again?

https://github.com/zephyrproject-rtos/zephyr/blob/af79664da3debec8d168e01dac6098dac88841cf/drivers/display/display_ili9xxx.c#L285-L292

Reply:
When rotate display, must swap X and Y

Reply:
this is already done by the driver shown above

Reply:
This is done by the driver, but not done by the LVGL layer.
If we have a 120x180 LVGL display, it will still draw, 120x180 a picture, but we need 180x120
See kpoint scan, same logic for changing X and Y
https://github.com/zephyrproject-rtos/zephyr/blob/10c971a463b83eebdcf7b0d4edc8bb8df5b8a4f5/lib/gui/lvgl/lvgl.c#L262-L267